src/main/java/apaintus/controllers/FigureLogController.java

```diff
@@ -0,0 +1,101 @@
+package apaintus.controllers;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.util.Callback;
+
+import java.util.List;
+
+import apaintus.models.ListViewCell;
+import apaintus.models.commands.Invoker;
+import apaintus.models.commands.SelectCommand;
+import apaintus.models.shapes.DrawableShape;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+
+public class FigureLogController implements ChildController<Controller> {
+	private Controller controller;
+	private CanvasController canvasController;
+	private Invoker invoker;
+
+	public static final ObservableList<String> observableListFigures = FXCollections.observableArrayList();
+
+	@FXML
+	private ListView<String> figureList;
+	private List<DrawableShape> shapeList;
+
+	private static final int UP = -1;
+	private static final int DOWN = 1;
+
+	@Override
+	public void injectParentController(Controller controller) {
+		this.controller = controller;
+		this.canvasController = this.controller.getCanvasController();
+		invoker = controller.getInvoker();
+	}
+
+	@Override
+	public void initialize() {
+		figureList.setOnMouseClicked(Event -> {
+			int indexOfSelectedItem = figureList.getSelectionModel().getSelectedIndex();
+			if (indexOfSelectedItem < 0)
```

Comment:
is this even possible?

Reply:
yes, in case you click but drag your mouse away the index item returned by the selection model is null and there return value of the index returned is -1